src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -273,6 +273,392 @@ ON dbo.CompartmentAssignment
 WHERE IsHistory = 0
 WITH (DATA_COMPRESSION = PAGE)
 
+GO
+
+/*************************************************************
+    Reference Search Param
+**************************************************************/
+
+CREATE TYPE dbo.ReferenceSearchParamTableType_1 AS TABLE  
+(
+    SearchParamId smallint NOT NULL,
+    BaseUri varchar(128) NULL,
+    ReferenceResourceTypeId smallint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL,
+    ReferenceResourceVersion int NULL
+)
+
+CREATE TABLE dbo.ReferenceSearchParam
+(
+    ResourceSurrogateId bigint NOT NULL,
+    SearchParamId smallint NOT NULL,
+    BaseUri varchar(128) NULL,
+    ReferenceResourceTypeId smallint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL,
+    ReferenceResourceVersion int NULL,
+    IsHistory bit NOT NULL,
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE CLUSTERED INDEX IXC_ReferenceSearchParam
+ON dbo.ReferenceSearchParam
+(
+    ResourceSurrogateId,
+    SearchParamId,
+    ReferenceResourceTypeId,
```

Comment:
Going to move the TypeId later in the indexes for this table, because you can perform a search without specifying the type. @jackliums, I don't think we handle this part of the spec:
> Servers SHOULD reject a search where the logical id refers to more than one matching resource across different types

Reply:
That's not yet implemented. We do have the metadata about the target resource type so we should be able to implement this pretty easily.